pandas/io/excel.py

```diff
@@ -312,11 +327,6 @@ def _range2cols(areas):
             >>> _range2cols('A,C,Z:AB')
             [0, 2, 25, 26, 27]
             """
-            def _excel2num(x):
-                "Convert Excel column name like 'AB' to 0-based column index"
-                return reduce(lambda s, a: s * 26 + ord(a) - ord('A') + 1,
```

Comment:
code changed to allow passing list of strings, similar to other `read_*` functions, and keep the `read_excel` "superpower" to pass a string that is parsed as mentioned in the original issue (#4988, point 3).